test/core/auxiliary_block_store/is_vote_valid.js

```diff
@@ -245,228 +213,579 @@ contract('AuxiliaryBlockStore.isVoteValid()', async (accounts) => {
+                        source: testBlocks[6].hash,
+                        target: testBlocks[10].hash,
+                        accumulatedGas: testBlocks[6].accumulatedGas,
+                        accumulatedTransactionRoot: testBlocks[6].accumulatedTransactionRoot,
+                    },
+                    {
+                        dynasty: new BN('3'),
+                        source: testBlocks[6].hash,
+                        target: testBlocks[12].hash,
+                        accumulatedGas: testBlocks[6].accumulatedGas,
+                        accumulatedTransactionRoot: testBlocks[6].accumulatedTransactionRoot,
+                    },
+                ],
+                activeKernel,
+            );
+
+            await pollingPlace.justify(
+                testBlocks[6].hash,
+                testBlocks[8].hash,
+            );
+
+            activeKernel = openKernel;
+            openKernel = web3.utils.sha3('openKernel5');
+            await kernelGateway.setOpenKernelHash(openKernel);
+            await kernelGateway.setOpenKernelActivationHeight(5);
+
+            await AuxStoreUtils.testValidVotes(
+                blockStore,
+                coreIdentifier,
+                [
+                    {
+                        dynasty: new BN('4'),
+                        source: testBlocks[8].hash,
+                        target: testBlocks[10].hash,
+                        accumulatedGas: testBlocks[8].accumulatedGas,
+                        accumulatedTransactionRoot: testBlocks[8].accumulatedTransactionRoot,
+                    },
+                    {
+                        dynasty: new BN('4'),
+                        source: testBlocks[8].hash,
+                        target: testBlocks[12].hash,
+                        accumulatedGas: testBlocks[8].accumulatedGas,
+                        accumulatedTransactionRoot: testBlocks[8].accumulatedTransactionRoot,
+                    },
+                ],
+                activeKernel,
+            );
+
+            await pollingPlace.justify(
+                testBlocks[8].hash,
+                testBlocks[10].hash,
+            );
+
+            activeKernel = openKernel;
+            openKernel = web3.utils.sha3('openKernel6');
+            await kernelGateway.setOpenKernelHash(openKernel);
+            await kernelGateway.setOpenKernelActivationHeight(6);
+
+
+            await AuxStoreUtils.testValidVotes(
+                blockStore,
+                coreIdentifier,
+                [
+                    {
+                        dynasty: new BN('5'),
+                        source: testBlocks[10].hash,
+                        target: testBlocks[12].hash,
+                        accumulatedGas: testBlocks[10].accumulatedGas,
+                        accumulatedTransactionRoot: testBlocks[10].accumulatedTransactionRoot,
+                    },
+                ],
+                activeKernel,
+            );
+
+        });
```

Comment:
Is it correct that these changes are part of your PR? 😕

Reply:
This test to check that votes pick up proper kernel hash once they get justified.